File: src/main/java/toker/warbandscripts/panel/controller/rest/BanController.java
package toker.warbandscripts.panel.controller.rest;

import com.fasterxml.jackson.annotation.JsonView;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import toker.warbandscripts.panel.entity.Ban;
import toker.warbandscripts.panel.service.BanService;

import java.util.List;

@RestController
public class BanController {

    private BanService banService;

    public BanController(BanService banService) {
        this.banService = banService;
    }

    @GetMapping("/api/bans/player/{playerId}")
    @JsonView(Ban.View.PanelUser.class)
    public List<Ban> bansByPlayer(@PathVariable int playerId) {
        return banService.getBansOfPlayer(playerId);
    }

    @GetMapping("/api/bans/admin/{adminId}")
    @JsonView(Ban.View.None.class)
    public List<Ban> bansByAdmin(@PathVariable int adminId) {
        return banService.getBansOfAdmin(adminId);
    }

    @PostMapping("/api/ban")
    @JsonView(Ban.View.None.class)
    public Ban ban(@RequestBody Ban ban) {
        return banService.banPlayer(ban);
    }

    @PutMapping("/api/unban/{banId}")
    public void undoBan(@PathVariable int banId) {
        banService.undoBan(banId);
    }

    @PutMapping("/api/unban/player/{playerUniqueId}")
    public void unbanPlayer(@PathVariable int playerUniqueId) {
        banService.undoAllBansForPlayer(playerUniqueId);
    }
}

File: src/main/java/toker/panel/controller/rest/BanController.kt
package toker.panel.controller.rest

import com.fasterxml.jackson.annotation.JsonView
import org.springframework.data.crossstore.ChangeSetPersister
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import org.springframework.web.server.ResponseStatusException
import toker.panel.entity.Ban
import toker.panel.service.BanService

@RestController
class BanController(private val banService: BanService) {
    @GetMapping("/api/players/{playerId}/bans")
    @JsonView(Ban.View.PanelUser::class)
    fun bansByPlayer(@PathVariable playerId: Int): List<Ban>? {
        return banService.getBansOfPlayer(playerId)
    }

    @GetMapping("/api/admins/{adminId}/bans")
    @JsonView(Ban.View.None::class)
    fun bansByAdmin(@PathVariable adminId: Int): List<Ban> {
        return banService.getBansOfAdmin(adminId)
    }

    @PostMapping("/api/bans")
    @JsonView(Ban.View.PanelUser::class)
    fun ban(@RequestBody ban: Ban): Ban {
        if (ban.minutes!! < 0) throw ResponseStatusException(HttpStatus.BAD_REQUEST)
        return banService.banPlayer(ban)
    }

    @PutMapping("/api/bans/{banId}/undo")
    @JsonView(Ban.View.None::class)
    @Throws(ChangeSetPersister.NotFoundException::class)
    fun undoBan(@PathVariable banId: Int): Ban {
        return banService.undoBan(banId)
    }

    @PutMapping("/api/bans/byUniqueId/{playerUniqueId}/undo")
    fun unbanPlayer(@PathVariable playerUniqueId: Int) {
        banService.undoAllBansForPlayer(playerUniqueId)
    }
}